lib/validator.js

```diff
@@ -291,9 +289,10 @@ function checkSubSchema(schema, parent, state) {
 
     if (schema.externalDocs) {
         schema.externalDocs.should.have.key('url');
-        should.doesNotThrow(function() {
- validateUrl(schema.externalDocs.url, [state.openapi.servers], 'externalDocs', state.options)
```

Comment:
This was an array wrapping an array (`[state.openapi.servers]`) which seemed a bit funny? #51 resulted in this being `[undefined]`, which was causing the error for @iainmcgin. This change makes sure it only sends servers through if they aren't undefined. 

dunno if this is any good. I invoke @MikeRalphson for this one.